Create and connect dynamic inputs in a single call

j-o opened the pull request:
This adds corresponding `Stage::createInput` overloads. Simplifies creating dynamic inputs from
```
*(m_programStage->createInput<cppassist::FilePath>("shader1")) << shader1;
*(m_programStage->createInput<globjects::Shader *>("shader2")) << m_shaderStage->shader;
```
to
```
m_programStage->createInput("shader1", shader1);
m_programStage->createInput("shader2", m_shaderStage->shader);
```

I considered adding similar overloads for `createOutput` for symmetry; however, I don't really see the use case for creating directly connected dynamic outputs?

Sorry for not unifying this with #254.

sbusch42 commented:
If would find it more intuitive if the function would be renamed into e.g. createInputFor() or createAndConnect(), to make it more clear that it does two things at once.

Second question: Wouldn't it be possible to unite the two functions into one, using Slot<T> instead of Input<T> and Output<T>?

scheibel commented:
I see one major advantage of your proposed code: template arguments are deduced from immediate input/output.

Is it possible to provide an implementation for the following, similar interface that uses the syntax of the DSL instead?
```cpp
m_programStage->createInput("shader1") << shader1;
m_programStage->createInput("shader2") << m_shaderStage->shader;
```
I suggest that ```createInput``` without type create a dummy object that has an ```operator<<``` overload and executes the actual creation and connection of the input there.
This extra object should use the functionality and interfaces suggested by @sbusch42

j-o commented:
Renamed to `createConnectedInput`. Also implemented @scheibel's suggestion.

The two separate overloads for Input and Output were required before, as otherwise, the base `createInput<T>` function was selected during overload resolution, resulting in `Input<Input<T>>`.
